modules/ROOT/pages/http/http-listener-ref.adoc

```diff
@@ -1,182 +1,296 @@
-= HTTP Listener Configuration Reference
+= HTTP Listener Reference
 ifndef::env-site,env-github[]
 include::_attributes.adoc[]
 endif::[]
 :keywords: anypoint studio, esb, connectors, http, https, http headers, query parameters, rest, raml
 
-The required minimum settings for the Listener operation are:
+The HTTP listener is a source that allows setting up an HTTP server and triggering flows when HTTP requests are received.
 
-* Host
-* Port
-* Path
+You can choose what methods the source accepts, such as GET, POST or a list with both, and on which path requests will be accepted on, therefore allowing routing the requests through different flows.
 
-You can choose what methods the connector accepts, such as GET or POST. Provide a comma-separated list of the methods that the connector allows in Advanced > Allowed Methods. The default blank list allows all methods. The connector routes incoming requests based on the URL that requests the app.
+Once a request is accepted by the source, the corresponding flow is triggered with the HTTP body as payload and the HTTP metadata as attributes (headers, query params and so on).
 
-You can configure the HTTP Listener to accept any request, or to accept only those requests addressed to a specific URI. You set the Path to limit the requests.
+When the flow finishes its execution, the HTTP listener allows customizing the HTTP response based on whether the execution was successful or not, so that different status codes may be returned, for example.
 
-The connector passes the request body to the next element of the Mule flow as the message payload of a Mule message. Headers, query parameters, and URI parameters are passed as attributes. You can access these properties from your Mule application using DataWeave.
+== HTTP Listener Configuration
 
-In response to the request that the Listener receives, you can configure response codes and clarifications for success and error outcomes to present different error messages depending on the outcome.
+To use an HTTP listener, a configuration must be declared with a corresponding connection. This establishes the HTTP server that will listen to requests.
 
-== Setting the Host
+Additionally, a base path can be configured that will apply to all listeners using the configuration.
 
-To deploy a project to CloudHub, setting Host = *0.0.0.0* is required.
+[source,xml,linenums]
+----
+<http:listener-config name="HTTP_Listener_config" basePath="api/v1">
+  <http:listener-connection host="0.0.0.0" port="8081" />
```

Comment:
since it's a common practice, should we add a comment around the difference between `0.0.0.0` and `localhost`. I know that this doc is about the http connector and not basic networking, but still......

Reply:
There's a tip around this on the connection section where this config is repeated.